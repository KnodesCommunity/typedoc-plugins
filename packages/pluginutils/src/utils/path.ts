// eslint-disable-next-line no-restricted-imports -- Re-export normalized
import { dirname as _dirname, join as _join, normalize as _normalize, relative as _relative, resolve as _resolve, basename, extname, isAbsolute, parse } from 'path';

import { isString } from 'lodash';

import { normalizePath } from 'typedoc';

export const resolve = ( from: string, ...to: Array<string | null | undefined> ) => normalizePath( _resolve( from, ...to.filter( isString ) ) );
export const join = ( ...paths: Array<string | null | undefined> ) => normalizePath( _join( ...paths.filter( isString ) ) );
export const relative = ( from: string, to: string ) => normalizePath( _relative( from, to ) );
export const dirname = ( path: string ) => normalizePath( _dirname( path ) );
export const normalize = ( path: string ) => normalizePath( _normalize( path ) );
export { isAbsolute, parse, basename, extname };
