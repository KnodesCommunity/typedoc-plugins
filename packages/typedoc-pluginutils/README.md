# typedoc-plugin-pages

> A TypeDoc plugin that lets you integrate your own pages into the documentation output

[![npm](https://img.shields.io/npm/v/typedoc-plugin-pages?color=brightgreen)](https://www.npmjs.com/package/typedoc-plugin-pages)

This plugin is based on [typedoc-plugin-loopingz-pages](https://github.com/loopingz/typedoc-plugin-loopingz-pages), which is in turn a fork of [typedoc-plugin-pages](https://github.com/mipatterson/typedoc-plugin-pages). Integrating it in this monorepo should (I hope) make easier maintainance.

## Compatibility

This plugin version should match TypeDoc `major.minor.x` for compatibility.

## Features

- 🔍 Search integration
- 🔗 Interpage hyperlinks
- 🎨 Integrated Theme
- 📑 Reflection Navigation Title Customization
- ✏️ Integrated spell-checking (coming soon...)

## Demo

You can view a live demo of the original `typedoc-plugin-pages` [here](https://mipatterson.github.io/typedoc-plugin-pages/).

## Usage

See the [Quick Start](https://mipatterson.github.io/typedoc-plugin-pages/pages/Getting%20Started/quick-start.html) guide to get started.

## Supported Versions of TypeDoc

This plugin is designed to work with as many versions of TypeDoc as possible. It has been tested with the following versions `0.16.5` through `0.17.6`. If you are reporting an issue, please include the version of TypeDoc you are using the plugin with.