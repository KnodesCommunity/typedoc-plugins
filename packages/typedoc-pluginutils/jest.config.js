const { resolve } = require( 'path' );

const base = require( '../../jest.config.base' );
/** @type {import('ts-jest/dist/types').InitialOptionsTsJest} */
module.exports = {
	...base,
	projects: [
		{
			...base.projects[0],
			setupFilesAfterEnv: [ ...base.projects[1].setupFilesAfterEnv, '@alex_neo/jest-expect-message' ],
		},
		{
			...base.projects[1],
			setupFilesAfterEnv: [ ...base.projects[1].setupFilesAfterEnv, '@testing-library/jest-dom' ],
		},
	],
	watchPathIgnorePatterns: [
		...( base.watchPathIgnorePatterns ?? [] ),
		resolve( __dirname, '__tests__/integration/mock-fs/.*' ),
	],
};
