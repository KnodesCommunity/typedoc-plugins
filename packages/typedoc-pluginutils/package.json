{
  "name": "@knodes/typedoc-pluginutils",
  "version": "0.0.1",
  "description": "A set of utilities for TypeDoc plugins",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/KnodesCommunity/typedoc-plugins.git"
  },
  "bugs": {
    "url": "https://github.com/KnodesCommunity/typedoc-plugins/issues"
  },
  "homepage": "https://github.com/KnodesCommunity/typedoc-plugins/tree/main/./packages/typedoc-pluginutils#readme",
  "author": {
    "email": "gerkin@knodes.org",
    "name": "GerkinDev"
  },
  "keywords": [
    "documentation",
    "plugin",
    "typedoc",
    "utilities",
    "utility",
    "utils"
  ],
  "main": "dist/index.js",
  "files": [
    "dist",
    "src"
  ],
  "scripts": {
    "build": "tsc --project tsconfig.build.json",
    "build:clean": "rimraf dist",
    "lint": "eslint --ignore-path ../../.eslintignore '**/?(.)*.?([cm])[tj]s?(x)'",
    "prepublish": "npm run build:clean && npm run build",
    "test": "jest --config jest.config.js"
  },
  "dependencies": {
    "lodash": "^4.17.0",
    "pkg-up": "^3.1.0",
    "semver": "^7.3.5"
  },
  "peerDependencies": {
    "lodash": "^4.17.0",
    "typedoc": "^0.22.0"
  },
  "devDependencies": {
    "@alex_neo/jest-expect-message": "^1.0.5",
    "@knodes/eslint-config": "^1.6.3",
    "@testing-library/jest-dom": "^5.16.2",
    "@types/jest": "^27.4.1",
    "@types/jsdom": "^16.2.14",
    "@types/lodash": "^4.14.178",
    "@types/node": "^17.0.21",
    "@types/react": "^17.0.39",
    "@types/semver": "^7.3.9",
    "@types/tmp": "^0.2.3",
    "@typescript-eslint/eslint-plugin": "^5.12.1",
    "@typescript-eslint/parser": "^5.12.1",
    "conventional-changelog-cli": "^2.2.2",
    "eslint": "^8.9.0",
    "eslint-plugin-eslint-comments": "^3.2.0",
    "eslint-plugin-import": "^2.25.4",
    "eslint-plugin-jsdoc": "^37.9.4",
    "eslint-plugin-prefer-arrow": "^1.2.3",
    "eslint-plugin-sort-export-all": "^1.2.1",
    "expect-type": "^0.13.0",
    "jest": "^27.5.1",
    "jest-extended": "^2.0.0",
    "jest-junit": "^13.0.0",
    "jsdom": "^19.0.0",
    "rimraf": "^3.0.2",
    "tmp": "^0.2.1",
    "ts-jest": "^27.1.3",
    "type-fest": "^2.12.0",
    "typedoc": "^0.22.12",
    "typescript": "^4.5.5"
  }
}
