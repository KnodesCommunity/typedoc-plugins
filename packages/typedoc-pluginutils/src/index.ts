export * from './autoload';
export * from './base-plugin';
export * from './current-page-memo';
export * from './markdown-replacer';
export * from './option';
export * from './path-reflection-resolver';
export * from './plugin-logger';
export * from './utils';
