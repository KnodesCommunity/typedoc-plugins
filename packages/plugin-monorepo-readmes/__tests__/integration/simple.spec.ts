import { resolve } from '@knodes/typedoc-pluginutils/path';

import { describeDocsFile, formatHtml, runPluginBeforeAll } from '#plugintestbed';

const rootDir = resolve( __dirname, '../mock-fs/simple' );
process.chdir( rootDir );
runPluginBeforeAll( rootDir, resolve( __dirname, '../../src/index' ) );
describe( 'Root `index.html`', describeDocsFile( rootDir, 'index.html', withContent => {
	it( 'should have correct content', withContent( ( content, dom ) => {
		const heading = dom.window.document.querySelectorAll( '.tsd-panel h1' );
		expect( heading ).toHaveLength( 1 );
		expect( heading[0] ).toHaveTextContent( 'Root readme' );
		expect( formatHtml( content ) ).toMatchSnapshot();
	} ) );
} ) );
describe( 'packages/a module`', describeDocsFile( rootDir, 'modules/_example_package_a.html', withContent => {
	it.failing( 'should have correct content', withContent( ( content, dom ) => {
		const heading = dom.window.document.querySelectorAll( '.tsd-panel h1' );
		expect( heading ).toHaveLength( 1 );
		expect( heading[0] ).toHaveTextContent( 'Readme of A' );
		expect( formatHtml( content ) ).toMatchSnapshot();
	} ) );
} ) );
describe( 'packages/b module`', describeDocsFile( rootDir, 'modules/_example_package_b.html', withContent => {
	it.failing( 'should have correct content', withContent( ( content, dom ) => {
		const heading = dom.window.document.querySelectorAll( '.tsd-panel h1' );
		expect( heading ).toHaveLength( 1 );
		expect( heading[0] ).toHaveTextContent( 'Readme of B' );
		expect( formatHtml( content ) ).toMatchSnapshot();
	} ) );
} ) );
