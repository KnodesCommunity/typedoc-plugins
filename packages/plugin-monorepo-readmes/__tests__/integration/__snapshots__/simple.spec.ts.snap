// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root module \`index.html\` should be correct 1`] = `
"<!DOCTYPE html>
<html class=\\"default\\">

<head>
	<meta charSet=\\"utf-8\\" />
	<meta http-equiv=\\"x-ua-compatible\\" content=\\"IE=edge\\" />
	<title>@example/my-packages</title>
	<meta name=\\"description\\" content=\\"Documentation for @example/my-packages\\" />
	<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\" />
	<link rel=\\"stylesheet\\" href=\\"assets/style.css\\" />
	<link rel=\\"stylesheet\\" href=\\"assets/highlight.css\\" />
	<script async src=\\"assets/search.js\\" id=\\"search-script\\"></script>
</head>

<body>
	<script>
		document.body.classList.add(localStorage.getItem(\\"tsd-theme\\") || \\"os\\")
	</script>
	<header>
		<div class=\\"tsd-page-toolbar\\">
			<div class=\\"container\\">
				<div class=\\"table-wrap\\">
					<div class=\\"table-cell\\" id=\\"tsd-search\\" data-base=\\".\\">
						<div class=\\"field\\"><label for=\\"tsd-search-field\\" class=\\"tsd-widget search no-caption\\">Search</label><input type=\\"text\\" id=\\"tsd-search-field\\" /></div>
						<ul class=\\"results\\">
							<li class=\\"state loading\\">Preparing search index...</li>
							<li class=\\"state failure\\">The search index is not available</li>
						</ul><a href=\\"index.html\\" class=\\"title\\">@example/my-packages</a>
					</div>
					<div class=\\"table-cell\\" id=\\"tsd-widgets\\">
						<div id=\\"tsd-filter\\"><a href=\\"#\\" class=\\"tsd-widget options no-caption\\" data-toggle=\\"options\\">Options</a>
							<div class=\\"tsd-filter-group\\">
								<div class=\\"tsd-select\\" id=\\"tsd-filter-visibility\\"><span class=\\"tsd-select-label\\">All</span>
									<ul class=\\"tsd-select-list\\">
										<li data-value=\\"public\\">Public</li>
										<li data-value=\\"protected\\">Public/Protected</li>
										<li data-value=\\"private\\" class=\\"selected\\">All</li>
									</ul>
								</div> <input type=\\"checkbox\\" id=\\"tsd-filter-inherited\\" checked /><label class=\\"tsd-widget\\" for=\\"tsd-filter-inherited\\">Inherited</label><input type=\\"checkbox\\" id=\\"tsd-filter-externals\\" checked /><label class=\\"tsd-widget\\" for=\\"tsd-filter-externals\\">Externals</label>
							</div>
						</div><a href=\\"#\\" class=\\"tsd-widget menu no-caption\\" data-toggle=\\"menu\\">Menu</a>
					</div>
				</div>
			</div>
		</div>
		<div class=\\"tsd-page-title\\">
			<div class=\\"container\\">
				<h1> @example/my-packages </h1>
			</div>
		</div>
	</header>
	<div class=\\"container container-main\\">
		<div class=\\"row\\">
			<div class=\\"col-8 col-content\\">
				<div class=\\"tsd-panel tsd-typography\\">
					<a href=\\"#root-readme\\" id=\\"root-readme\\" style=\\"color: inherit; text-decoration: none;\\">
						<h1>Root readme</h1>
					</a>
				</div>
			</div>
			<div class=\\"col-4 col-menu menu-sticky-wrap menu-highlight\\">
				<nav class=\\"tsd-navigation primary\\">
					<ul>
						<li class=\\"current\\"><a href=\\"modules.html\\">Modules</a></li>
						<li class=\\" tsd-kind-module\\"><a href=\\"modules/_example_package_a.html\\">@example/package-<wbr />a</a></li>
						<li class=\\" tsd-kind-module\\"><a href=\\"modules/_example_package_b.html\\">@example/package-<wbr />b</a></li>
					</ul>
				</nav>
			</div>
		</div>
	</div>
	<footer class=\\"with-border-bottom\\">
		<div class=\\"container\\">
			<h2>Legend</h2>
			<div class=\\"tsd-legend-group\\">
				<ul class=\\"tsd-legend\\">
					<li class=\\"tsd-kind-variable\\"><span class=\\"tsd-kind-icon\\">Variable</span></li>
				</ul>
			</div>
			<h2>Settings</h2>
			<p>Theme <select id=\\"theme\\">
					<option value=\\"os\\">OS</option>
					<option value=\\"light\\">Light</option>
					<option value=\\"dark\\">Dark</option>
				</select></p>
		</div>
	</footer>
	<div class=\\"container tsd-generator\\">
		<p>Generated using <a href=\\"https://typedoc.org/\\" target=\\"_blank\\">TypeDoc</a></p>
	</div>
	<div class=\\"overlay\\"></div>
	<script src=\\"assets/main.js\\"></script>
</body>

</html>"
`;

exports[`packages/a module \`modules/_example_package_a.html\` should be correct 1`] = `
"<!DOCTYPE html>
<html class=\\"default\\">

<head>
	<meta charSet=\\"utf-8\\" />
	<meta http-equiv=\\"x-ua-compatible\\" content=\\"IE=edge\\" />
	<title>@example/package-a | @example/my-packages</title>
	<meta name=\\"description\\" content=\\"Documentation for @example/my-packages\\" />
	<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\" />
	<link rel=\\"stylesheet\\" href=\\"../assets/style.css\\" />
	<link rel=\\"stylesheet\\" href=\\"../assets/highlight.css\\" />
	<script async src=\\"../assets/search.js\\" id=\\"search-script\\"></script>
</head>

<body>
	<script>
		document.body.classList.add(localStorage.getItem(\\"tsd-theme\\") || \\"os\\")
	</script>
	<header>
		<div class=\\"tsd-page-toolbar\\">
			<div class=\\"container\\">
				<div class=\\"table-wrap\\">
					<div class=\\"table-cell\\" id=\\"tsd-search\\" data-base=\\"..\\">
						<div class=\\"field\\"><label for=\\"tsd-search-field\\" class=\\"tsd-widget search no-caption\\">Search</label><input type=\\"text\\" id=\\"tsd-search-field\\" /></div>
						<ul class=\\"results\\">
							<li class=\\"state loading\\">Preparing search index...</li>
							<li class=\\"state failure\\">The search index is not available</li>
						</ul><a href=\\"../index.html\\" class=\\"title\\">@example/my-packages</a>
					</div>
					<div class=\\"table-cell\\" id=\\"tsd-widgets\\">
						<div id=\\"tsd-filter\\"><a href=\\"#\\" class=\\"tsd-widget options no-caption\\" data-toggle=\\"options\\">Options</a>
							<div class=\\"tsd-filter-group\\">
								<div class=\\"tsd-select\\" id=\\"tsd-filter-visibility\\"><span class=\\"tsd-select-label\\">All</span>
									<ul class=\\"tsd-select-list\\">
										<li data-value=\\"public\\">Public</li>
										<li data-value=\\"protected\\">Public/Protected</li>
										<li data-value=\\"private\\" class=\\"selected\\">All</li>
									</ul>
								</div> <input type=\\"checkbox\\" id=\\"tsd-filter-inherited\\" checked /><label class=\\"tsd-widget\\" for=\\"tsd-filter-inherited\\">Inherited</label><input type=\\"checkbox\\" id=\\"tsd-filter-externals\\" checked /><label class=\\"tsd-widget\\" for=\\"tsd-filter-externals\\">Externals</label>
							</div>
						</div><a href=\\"#\\" class=\\"tsd-widget menu no-caption\\" data-toggle=\\"menu\\">Menu</a>
					</div>
				</div>
			</div>
		</div>
		<div class=\\"tsd-page-title\\">
			<div class=\\"container\\">
				<ul class=\\"tsd-breadcrumb\\">
					<li><a href=\\"../modules.html\\">@example/my-packages</a></li>
					<li><a href=\\"_example_package_a.html\\">@example/package-a</a></li>
				</ul>
				<h1>Module @example/package-a </h1>
			</div>
		</div>
	</header>
	<div class=\\"container container-main\\">
		<div class=\\"row\\">
			<div class=\\"col-8 col-content\\">
				<div class=\\"tsd-panel tsd-typography\\">
					<a href=\\"#readme-of-a\\" id=\\"readme-of-a\\" style=\\"color: inherit; text-decoration: none;\\">
						<h1>Readme of A</h1>
					</a>
				</div>
				<hr />
				<section class=\\"tsd-panel-group tsd-index-group\\">
					<h2>Index</h2>
					<section class=\\"tsd-panel tsd-index-panel\\">
						<div class=\\"tsd-index-content\\">
							<section class=\\"tsd-index-section \\">
								<h3>Variables</h3>
								<ul class=\\"tsd-index-list\\">
									<li class=\\"tsd-kind-variable tsd-parent-kind-module\\"><a href=\\"_example_package_a.html#A\\" class=\\"tsd-kind-icon\\">A</a></li>
								</ul>
							</section>
						</div>
					</section>
				</section>
				<section class=\\"tsd-panel-group tsd-member-group \\">
					<h2>Variables</h2>
					<section class=\\"tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module\\"><a id=\\"A\\" class=\\"tsd-anchor\\"></a>
						<h3 class=\\"tsd-anchor-link\\"><span class=\\"tsd-flag ts-flagConst\\">Const</span> A<a href=\\"#A\\" aria-label=\\"Permalink\\" class=\\"tsd-anchor-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"icon icon-tabler icon-tabler-link\\" viewBox=\\"0 0 24 24\\" stroke-width=\\"2\\" stroke=\\"currentColor\\" fill=\\"none\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\">
									<path stroke=\\"none\\" d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
									<path d=\\"M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5\\"></path>
									<path d=\\"M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5\\"></path>
								</svg></a></h3>
						<div class=\\"tsd-signature tsd-kind-icon\\">A<span class=\\"tsd-signature-symbol\\">:</span> <span class=\\"tsd-signature-type\\">&quot;a&quot;</span><span class=\\"tsd-signature-symbol\\"> = &#39;a&#39;</span></div>
						<aside class=\\"tsd-sources\\">
							<ul>
								<li>Defined in a/index.ts:1</li>
							</ul>
						</aside>
					</section>
				</section>
			</div>
			<div class=\\"col-4 col-menu menu-sticky-wrap menu-highlight\\">
				<nav class=\\"tsd-navigation primary\\">
					<ul>
						<li class=\\"\\"><a href=\\"../modules.html\\">Modules</a></li>
						<li class=\\"current tsd-kind-module\\"><a href=\\"_example_package_a.html\\">@example/package-<wbr />a</a></li>
						<li class=\\" tsd-kind-module\\"><a href=\\"_example_package_b.html\\">@example/package-<wbr />b</a></li>
					</ul>
				</nav>
				<nav class=\\"tsd-navigation secondary menu-sticky\\">
					<ul>
						<li class=\\"tsd-kind-variable tsd-parent-kind-module\\"><a href=\\"_example_package_a.html#A\\" class=\\"tsd-kind-icon\\">A</a></li>
					</ul>
				</nav>
			</div>
		</div>
	</div>
	<footer class=\\"with-border-bottom\\">
		<div class=\\"container\\">
			<h2>Legend</h2>
			<div class=\\"tsd-legend-group\\">
				<ul class=\\"tsd-legend\\">
					<li class=\\"tsd-kind-variable\\"><span class=\\"tsd-kind-icon\\">Variable</span></li>
				</ul>
			</div>
			<h2>Settings</h2>
			<p>Theme <select id=\\"theme\\">
					<option value=\\"os\\">OS</option>
					<option value=\\"light\\">Light</option>
					<option value=\\"dark\\">Dark</option>
				</select></p>
		</div>
	</footer>
	<div class=\\"container tsd-generator\\">
		<p>Generated using <a href=\\"https://typedoc.org/\\" target=\\"_blank\\">TypeDoc</a></p>
	</div>
	<div class=\\"overlay\\"></div>
	<script src=\\"../assets/main.js\\"></script>
</body>

</html>"
`;

exports[`packages/b module \`modules/_example_package_b.html\` should be correct 1`] = `
"<!DOCTYPE html>
<html class=\\"default\\">

<head>
	<meta charSet=\\"utf-8\\" />
	<meta http-equiv=\\"x-ua-compatible\\" content=\\"IE=edge\\" />
	<title>@example/package-b | @example/my-packages</title>
	<meta name=\\"description\\" content=\\"Documentation for @example/my-packages\\" />
	<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\" />
	<link rel=\\"stylesheet\\" href=\\"../assets/style.css\\" />
	<link rel=\\"stylesheet\\" href=\\"../assets/highlight.css\\" />
	<script async src=\\"../assets/search.js\\" id=\\"search-script\\"></script>
</head>

<body>
	<script>
		document.body.classList.add(localStorage.getItem(\\"tsd-theme\\") || \\"os\\")
	</script>
	<header>
		<div class=\\"tsd-page-toolbar\\">
			<div class=\\"container\\">
				<div class=\\"table-wrap\\">
					<div class=\\"table-cell\\" id=\\"tsd-search\\" data-base=\\"..\\">
						<div class=\\"field\\"><label for=\\"tsd-search-field\\" class=\\"tsd-widget search no-caption\\">Search</label><input type=\\"text\\" id=\\"tsd-search-field\\" /></div>
						<ul class=\\"results\\">
							<li class=\\"state loading\\">Preparing search index...</li>
							<li class=\\"state failure\\">The search index is not available</li>
						</ul><a href=\\"../index.html\\" class=\\"title\\">@example/my-packages</a>
					</div>
					<div class=\\"table-cell\\" id=\\"tsd-widgets\\">
						<div id=\\"tsd-filter\\"><a href=\\"#\\" class=\\"tsd-widget options no-caption\\" data-toggle=\\"options\\">Options</a>
							<div class=\\"tsd-filter-group\\">
								<div class=\\"tsd-select\\" id=\\"tsd-filter-visibility\\"><span class=\\"tsd-select-label\\">All</span>
									<ul class=\\"tsd-select-list\\">
										<li data-value=\\"public\\">Public</li>
										<li data-value=\\"protected\\">Public/Protected</li>
										<li data-value=\\"private\\" class=\\"selected\\">All</li>
									</ul>
								</div> <input type=\\"checkbox\\" id=\\"tsd-filter-inherited\\" checked /><label class=\\"tsd-widget\\" for=\\"tsd-filter-inherited\\">Inherited</label><input type=\\"checkbox\\" id=\\"tsd-filter-externals\\" checked /><label class=\\"tsd-widget\\" for=\\"tsd-filter-externals\\">Externals</label>
							</div>
						</div><a href=\\"#\\" class=\\"tsd-widget menu no-caption\\" data-toggle=\\"menu\\">Menu</a>
					</div>
				</div>
			</div>
		</div>
		<div class=\\"tsd-page-title\\">
			<div class=\\"container\\">
				<ul class=\\"tsd-breadcrumb\\">
					<li><a href=\\"../modules.html\\">@example/my-packages</a></li>
					<li><a href=\\"_example_package_b.html\\">@example/package-b</a></li>
				</ul>
				<h1>Module @example/package-b </h1>
			</div>
		</div>
	</header>
	<div class=\\"container container-main\\">
		<div class=\\"row\\">
			<div class=\\"col-8 col-content\\">
				<div class=\\"tsd-panel tsd-typography\\">
					<a href=\\"#readme-of-b\\" id=\\"readme-of-b\\" style=\\"color: inherit; text-decoration: none;\\">
						<h1>Readme of B</h1>
					</a>
				</div>
				<hr />
				<section class=\\"tsd-panel-group tsd-index-group\\">
					<h2>Index</h2>
					<section class=\\"tsd-panel tsd-index-panel\\">
						<div class=\\"tsd-index-content\\">
							<section class=\\"tsd-index-section \\">
								<h3>Variables</h3>
								<ul class=\\"tsd-index-list\\">
									<li class=\\"tsd-kind-variable tsd-parent-kind-module\\"><a href=\\"_example_package_b.html#B\\" class=\\"tsd-kind-icon\\">B</a></li>
								</ul>
							</section>
						</div>
					</section>
				</section>
				<section class=\\"tsd-panel-group tsd-member-group \\">
					<h2>Variables</h2>
					<section class=\\"tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module\\"><a id=\\"B\\" class=\\"tsd-anchor\\"></a>
						<h3 class=\\"tsd-anchor-link\\"><span class=\\"tsd-flag ts-flagConst\\">Const</span> B<a href=\\"#B\\" aria-label=\\"Permalink\\" class=\\"tsd-anchor-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"icon icon-tabler icon-tabler-link\\" viewBox=\\"0 0 24 24\\" stroke-width=\\"2\\" stroke=\\"currentColor\\" fill=\\"none\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\">
									<path stroke=\\"none\\" d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
									<path d=\\"M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5\\"></path>
									<path d=\\"M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5\\"></path>
								</svg></a></h3>
						<div class=\\"tsd-signature tsd-kind-icon\\">B<span class=\\"tsd-signature-symbol\\">:</span> <span class=\\"tsd-signature-type\\">&quot;b&quot;</span><span class=\\"tsd-signature-symbol\\"> = &#39;b&#39;</span></div>
						<aside class=\\"tsd-sources\\">
							<ul>
								<li>Defined in b/index.ts:1</li>
							</ul>
						</aside>
					</section>
				</section>
			</div>
			<div class=\\"col-4 col-menu menu-sticky-wrap menu-highlight\\">
				<nav class=\\"tsd-navigation primary\\">
					<ul>
						<li class=\\"\\"><a href=\\"../modules.html\\">Modules</a></li>
						<li class=\\" tsd-kind-module\\"><a href=\\"_example_package_a.html\\">@example/package-<wbr />a</a></li>
						<li class=\\"current tsd-kind-module\\"><a href=\\"_example_package_b.html\\">@example/package-<wbr />b</a></li>
					</ul>
				</nav>
				<nav class=\\"tsd-navigation secondary menu-sticky\\">
					<ul>
						<li class=\\"tsd-kind-variable tsd-parent-kind-module\\"><a href=\\"_example_package_b.html#B\\" class=\\"tsd-kind-icon\\">B</a></li>
					</ul>
				</nav>
			</div>
		</div>
	</div>
	<footer class=\\"with-border-bottom\\">
		<div class=\\"container\\">
			<h2>Legend</h2>
			<div class=\\"tsd-legend-group\\">
				<ul class=\\"tsd-legend\\">
					<li class=\\"tsd-kind-variable\\"><span class=\\"tsd-kind-icon\\">Variable</span></li>
				</ul>
			</div>
			<h2>Settings</h2>
			<p>Theme <select id=\\"theme\\">
					<option value=\\"os\\">OS</option>
					<option value=\\"light\\">Light</option>
					<option value=\\"dark\\">Dark</option>
				</select></p>
		</div>
	</footer>
	<div class=\\"container tsd-generator\\">
		<p>Generated using <a href=\\"https://typedoc.org/\\" target=\\"_blank\\">TypeDoc</a></p>
	</div>
	<div class=\\"overlay\\"></div>
	<script src=\\"../assets/main.js\\"></script>
</body>

</html>"
`;
